### DISCUSSION TITLE
Syntax error in class definition

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/brighterscript#862] [Q&A]
I'm trying to define the following class.

```
class Bookmark

    iRegistry as object
    iRegistrySection as object

    '''''''''
    ' new: Class constructor.
    ' 
    ' @param {string} section - registry section name.
    '''''''''
    sub new(section as string)
        ' Create the registry object.
        m.iRegistry = CreateObject("roRegistry")
        m.iRegisstrySection = CreateObject("roRegistrySection", section)
    end sub
```

I get the following errors:

--- Syntax Error. (compile error &h02 (ERR_SYNTAX): syntax error ) in pkg:/components/UILogic/BookmarkLogic.bs(4)
--- Syntax Error. (compile error &h02 (ERR_SYNTAX): syntax error ) in pkg:/components/UILogic/BookmarkLogic.bs(6)
--- Syntax Error. (compile error &h02 (ERR_SYNTAX): syntax error ) in pkg:/components/UILogic/BookmarkLogic.bs(7)
--- Syntax Error. (compile error &h02 (ERR_SYNTAX): syntax error ) in pkg:/components/UILogic/BookmarkLogic.bs(14)
--- Syntax Error. (compile error &h02 (ERR_SYNTAX): syntax error ) in pkg:/components/UILogic/BookmarkLogic.bs(18)

What am I missing?

I should also mention that I'm getting these messages while trying to debug in Visual Studio Code.

**Comment by TwitchBronBron** [comment]
It looks like you're trying to upload the raw brightERscript code to a Roku device. 

BrighterScript requires a transpile (compile) step before it can be run on a Roku. You need to run `npx bsc` to build your code into the ./out folder. The contents there is the code you should upload to Roku. 

Check out this sample project for an example that works in vscode. 

https://github.com/rokucommunity/brighterscript-template

**Reply by rrirower** [reply]
Thanks for the reply. A few more questions....

Can I execute the command you posted from within vscode?

Will I still be able to step through code using vscode?

Are there any extra files that need to be distributed with my code to support brighterscript?

**Comment by TwitchBronBron** [comment]
> Can I execute the command you posted from within vscode?

> Will I still be able to step through code using vscode

Yes. Look at the sample project I posted, it sets up vscode for exactly these reasons. Look at  preLaunchTask in launch.json and "build" in tasks.json.. you should pull down the sample project and test it out. 

> Are there any extra files that need to be distributed with my code to support brighterscript?

The output folder generated by `npx bsc` has your entire project and any extra required files, so just copy the entire folder and you'll be good.

**Comment by rrirower** [comment]
Thanks for the template project.  It was helpful.  I have a couple of observations.  Please let me know if I'm doing something wrong.

-  I had to run "npm install brighterscript -g"
-  I copied your launch.json and tasks.json files to my corresponding .vscode folder.
-  For debugging, I had to unzip the created zip file and use the code within it instead of my original source folders.
-  There appears to be only one extra file (bslib.brs) that I need to include with the channel code.

Is the above a correct way to proceed?  I am developing on Windows.

**Comment by TwitchBronBron** [comment]
I can write a longer explanation in a day or two, but in the meantime you can watch this tutorial on how to convert a project to BrighterScript, it might help answer a few of your questions. 

https://youtu.be/1rjNGi9c-rA

**Reply by rrirower** [reply]
Thanks, I'll take a look at it.

**Reply by rrirower** [reply]
I viewed the video.  However, as was stated in the video, the part I needed the most, the "setup", was glossed over quickly, so, I did not get too much from it.  I'm still having to switch back and forth within vscode when I want to update the source and debug from the output folder.  My workaround is to run "bsc" from a command prompt before debugging the code.

**Reply by rrirower** [reply]
I finally got the setup correct after digging a little deeper.  It wasn't obvious how to set up the launch.json, package.json, and bsconfig.json files.  Setting them up correctly made the difference.  I can now debug my code from within vscode without having to run "bsc" externally.  Perhaps a wiki post regarding how to set this up on Windows could help other users.
